### PR TITLE
Depend on @bazel_tools//third_party/ijar:zip from java_tools

### PR DESCRIPTION
**Pull request by meteorcloudy**
Fixes https://github.com/bazelbuild/bazel/issues/8614

### CODE REVIEW

File: third_party/ijar/BUILD [hunk 1]
```diff
@@ -30,6 +30,7 @@ cc_library(
         "//src:__subpackages__",
         "//third_party/ijar:__subpackages__",
         "//tools/test:__pkg__",
+        "@local_java_tools//:__pkg__",
```
Comment (iirina) [inline third_party/ijar/BUILD]: I'm not sure this is enough. It's enough for the tests to pass. The remote repo names in the workspace suffix are "remote_java_tools_linux", "remote_java_tools_windows" and "remote_java_tools_darwin". Can you please add them as well? Also add a TODO to migrate this target out of bazel_tools.

**Comment by meteorcloudy** [comment]
@iirina Thanks! I added tests and extra targets.

**Comment by meteorcloudy** [comment]
I'm not quite sure the test will pass because of
https://github.com/bazelbuild/bazel/blob/8ef17990f607b9e57d6d4336b807d329f3c0fff9/tools/jdk/BUILD.java_tools#L5-L6

We didn't declare rules_cc and rules_java anywhere in the WORKSPACE file.

**Comment by iirina** [comment]
Thanks @meteorcloudy!  `@rules_java` and `@rules_cc` are declared in the implicit workspace suffix: https://github.com/bazelbuild/bazel/blob/8ef17990f607b9e57d6d4336b807d329f3c0fff9/src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE#L242-L263

**Comment by meteorcloudy** [comment]
Building `singlejar_cc_bin` fails on all platform, looks like this is not a Windows only problem.

**Comment by iirina** [comment]
Thanks @meteorcloudy. This is something new, the tools were building before... Let me see what changed :(

**Comment by meteorcloudy** [comment]
OK, thanks for checking!

**Comment by iirina** [comment]
@meteorcloudy f5a6ee171dfa51dc06818591a9df5b5f469990be fixed the tests on the other platforms. Can you rebase and remove the lines that skip the tests on Windows?

https://github.com/bazelbuild/bazel/blob/f5a6ee171dfa51dc06818591a9df5b5f469990be/src/test/shell/bazel/bazel_java_tools_test.sh#L193-L195

and 

https://github.com/bazelbuild/bazel/blob/f5a6ee171dfa51dc06818591a9df5b5f469990be/src/test/shell/bazel/bazel_java_tools_test.sh#L212-L214

**Comment by meteorcloudy** [comment]
Found a better fix at https://github.com/bazelbuild/bazel/pull/8954, please take a look ;)

**Comment by meteorcloudy** [comment]
Closing due to https://github.com/bazelbuild/bazel/pull/8954

**Pull request closed without merging**